code/datums/components/qdeleting.dm

```diff
@@ -0,0 +1,41 @@
+/**
+ * A technical component which allows datums to register to other datums their QDELETION.
+ * Created so that, for instance, a human can listen to the same object twice to handle different things.
+ * Since you can only subscribe to a target once given a specific signal.
+ */
+
+/datum/component/qdeleting
```

Comment:
I was thinking of making a more generic component for this. One that accepts any signal.
But I decided against this for the reasons listed in the description. TLDR: Bad design in my mind